pkg/reconciler/config.go

```diff
@@ -69,6 +77,29 @@ type JetStreamVersion struct {
 	StartCommand         string `json:"startCommand"`
 }
 
+func (g *GlobalConfig) GetDefaultContainerResources() (*corev1.ResourceRequirements, error) {
+	// the standard resources used by the `init` and `main` containers.
+	defaultResources := corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{},
+		Requests: corev1.ResourceList{
+			"cpu":    resource.MustParse("100m"),
+			"memory": resource.MustParse("128Mi"),
+		},
+	}
+
+	if g.Defaults == nil || g.Defaults.ContainerResources == "" {
+		return &defaultResources, nil
+	}
+
+	var resourceConfig corev1.ResourceRequirements
+	err := yaml.Unmarshal([]byte(g.Defaults.ContainerResources), &resourceConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed unmarshal configuration file. %w", err)
```

Comment:
`panic(ftm.Errorf("failed to unmarshal default container resources, %w", err))`

In this way there's no need to return an error but a struct only, that will make the invokers easier - letting the reconciler capture the error and keep retrying does not make sense.